File: backend/internal/pkg/data/resource.go
package data

type ResourceService interface {
	ResourceByID(id int64) (Resource, error)
	CreateResource(data []byte, mimeType string, audioDuration int64) (Resource, error)
}

type Resource struct {
	ID            int64
	Data          []byte
	MIMEType      string
	AudioDuration int64
}

File: frontend/lib/api/model/resource.dart
import 'package:hive/hive.dart';

part 'resource.g.dart';

@HiveType(typeId: 2)
class Resource extends HiveObject {
  @HiveField(0)
  final int id;

  @HiveField(1)
  final String data;

  @HiveField(2)
  final String mimeType;

  @HiveField(3)
  final Duration? audioDuration;

  Resource({
    required this.id,
    required this.data,
    required this.mimeType,
    required this.audioDuration,
  });
}
